### ISSUE TITLE
gradient ascent

### ISSUE THREAD

**Issue by Vanitha83** [Vanitha83/Decimation#2]
Hi Jacob,

@jmschrei, I am maximizing a likelihood function which  should increase for small values of learning rate(R). But in my calculation, it increases and then decreases. The likehood function has logarithmic that always gives negative values for the gradient .  In addition, the state of the node ```ss[k]``` goes to zero when it is susceptible in the network, that ends up with the chosen initial coupling value ``` L_old[i][k]```.  Link to the equation that I am trying to solve is here: 

https://github.com/Vanitha83/Decimation/blob/master/discussion.pdf
```
for(i = 0, k = 1, t = 0.0; i < NN-1; i++)
 {
     L[i][k] = (x1[i]*ss[k])-(pow((1-x[i]),sum1)/(1-pow((1-x[i]),sum1)))*(x1[i]*ss[k]);
     L1[i][k] = L_old[i][k]+(R*L[i][k]);
}
```
Even in the presence of log can one expect that the error rate will increase as a function of time step?

**Comment by jmschrei** [comment]
Howdy. Are you sure you tagged the right person?

**Comment by Vanitha83** [comment]
Yes. As you are an expert in machine learning problems, I thought that you may give me some suggestion on this specific gradient ascent problem.

**Comment by Vanitha83** [comment]
Yes, but I just found that the log scale in my calculation decreases the error at first and then increases for different time step whatever be the choice of learning rate.

**Comment by jmschrei** [comment]
The error of what? If it's the error of your validation/test set, that's a sign that you are overfitting to your training data. If it's the error of your training set, that indicates you may have too large a learning rate that can't converge. I really have no context for any of these questions though so can't provide a more complete answer, sorry!

**Comment by Vanitha83** [comment]
okay..Error of the test set. Let me look into the calculation in the coding.
